python_modules/dagster/dagster/core/definitions/partition.py

```diff
@@ -495,7 +495,9 @@ def run_config_for_partition(self, partition: Partition[T]) -> Dict[str, Any]:
         return copy.deepcopy(self._user_defined_run_config_fn_for_partition(partition))
 
     def tags_for_partition(self, partition: Partition[T]) -> Dict[str, str]:
-        user_tags = copy.deepcopy(self._user_defined_tags_fn_for_partition(partition))
+        user_tags = copy.deepcopy(
+            validate_tags(self._user_defined_tags_fn_for_partition(partition))
+        )
         check_tags(user_tags, "user_tags")
```

Comment:
do we still need check_tags ?

Reply:
yeah there are some USER_EDITABLE_SYSTEM_TAGS checks in there (although these two methods are pretty confusing and could arguably be merged..)